course-template-bundle-spec.md

```diff
@@ -182,3 +183,16 @@ activity_options | ✓        | array       | `activity_options` is an array of
 -- id            | ✓        | string      | Reference to the unique identifier for the activity - `library/slug`.
 prompt           |          | dictionary  | Key is `locales` and each locale is a dictionary mapping locale codes to a prompt describing the step
 optional         |          | boolean     | `true` if the step is *not* required for completion
+
+### Retake Policies
+
+A retake policy defines any required cooldown periods, retake limits, and retake windows applicable to a given `CourseTemplate` activity.
+
+attribute               | required | type        | notes
+----------------------- | -------- | ----------- | -----------------------------------------
+id                      | ✓        | string      | A unique identifier for this retake policy
+activity_type           | ✓        | enum        | The type of retakeable activity to which this retake policy applies; one of `Lab` or `Quiz`
+activity_subtype        |          | string      | Distinguishes a subset of a given type of retakeable activity to which this retake policy applies (i.e. `graded` for quiz)
+retake_cooldown         |          | array       | An array of `n` integers (greater than or equal to 0) specifying the required cooldown periods (in days) between consequent retakes; the last integer will be the cooldown period for all retakes after the `nth`
+retake_limit            |          | integer     | The total number of attempts (greater than 0) allowed for the given type and subtype of retakeable activity; required if `retake_window` below is specified
+retake_window           |          | integer     | An integer (greater than 0) specifying the period (in days) for which the `retake_limit` above applies
```

Comment:
these are complicated enough that i think it would be nice to have a few examples

Reply:
Done.

Reply:
sorry, by examples i meant prose. like "for example, a retake policy of FOO means that a student can take a quiz up to N times per 24-hour period" or "for example, a retake policy of BAR means that after their first attempt, the student must wait X days before their second, and after their second attempt they must wait Y days before their third." ... stuff like that

Reply:
Ah, done.